haystack/nodes/preprocessor/splitter.py

```diff
@@ -0,0 +1,738 @@
+
+                            If the number of units is irrelevant, `split_length` can be safely set at 0.
+
+        :param add_page_number: Add the number of the page a paragraph occurs in to the Document's meta
+                                field `"page"`. Page boundaries are determined by `"\f"' character which is added
+                                in between pages by `PDFToTextConverter`, `TikaConverter`, `ParsrConverter` and
+                                `AzureConverter`.
+        """
+        documents = [
+            self.run(
+                documents=docs,
+                split_by=split_by,
+                split_regex=split_regex,
+                split_length=split_length,
+                split_overlap=split_overlap,
+                max_chars=split_max_chars,
+                add_page_number=add_page_number,
+            )[0]["documents"]
+            for docs in tqdm(documents, disable=not self.progress_bar, desc="Splitting", unit="docs")
+        ]
+        return {"documents": documents}, "output_1"
+
+    def split_into_units(
+        self, document: Document, units: Tuple[List[str], List[int]], add_page_number: bool = True
+    ) -> Tuple[List[Document], List[int]]:
+        """
+        Splits the parent document into single unit documents.
+
+        This algorithm is heavily simplified by the lack of split_length and split_overlap. Those are later applied by the merger.
+
+        :param document: The document to split.
+        :param units: Two parallel lists of (strings, offsets). Offsets is populated only if the split was performed on a regex
+            (so split_by "regex", "page", "paragraph" or "word", but not "sentence", "token" or "character"). In these cases,
+            offsets will contain the length of the regex matched to split. Is used by the DocumentCleaner to remove matches
+            without having to re-match the documents.
+            For example, if split_by="word":
+            - units=(['ab ', 'cd. \n\n', 'ef!'], [1, 3, 0]) means that there is one whitespace, three whitespaces and no whitespace
+                at the end of each of the strings.
+        :param add_page_number: If `True`, ounts the number of form feeds to assign to each split a metadata entry with the page where it starts
+                                in the original document.
+        """
+        if isinstance(document, dict):
+            warnings.warn(
+                "Use Document objects. Passing a dictionary to DocumentSplitter is deprecated.", DeprecationWarning
+            )
+            document = Document.from_dict(document)
+
+        if document.content_type != "text":
+            raise ValueError(
+                f"DocumentSplitter received a '{document.content_type}' document. "
+                "Make sure to pass only text documents to it. "
+                "You can use a RouteDocuments node to make sure only text document are sent to the DocumentCleaner."
+            )
```

Comment:
For consistency with `NewPreProcessor`, `DocumentMerger` and `DocumentCleaner`, should we move this check up to the `run` method?